src/field/gen_fieldops.py

```diff
@@ -185,10 +187,17 @@ def mymin(f1,f2):
             for d in dims:
                 print("  }")
             print("  }")
```

Comment:
I think this approach to balancing braces is a bit fragile, since changes to the generation code need be made in several places simultaneously. I have seen this done elsewhere using "with" blocks which emit open braces on entering, and closing braces on exit:

    with braces:
        some code

Reply:
It now has the disadvantage that the there is only one set of braces for the triple for loop - I hope that this is fine - as it is represented in python.